core/controllers/suggestion.py

```diff
@@ -83,13 +89,48 @@ class SuggestionHandler(base.BaseHandler):
 
     @acl_decorators.can_suggest_changes
     def post(self):
+        change_dict = self.payload.get('change')
+        target_id = self.payload.get('target_id')
+        target_type = self.payload.get('target_type')
+        html_list = []
+        if target_type == suggestion_models.TARGET_TYPE_SKILL:
+            state_dict = change_dict['question_dict']['question_state_data']
+            state = state_domain.State.from_dict(state_dict)
+            html_list = state.get_all_html_content_strings()
```

Comment:
Maybe create the suggestion object here and call get_all_html_list (Only save the Suggestion in the datastore at the end.)

Reply:
done